remote-api/src/main/java/org/alfresco/rest/api/nodes/NodesEntityResource.java

```diff
@@ -225,5 +275,116 @@ public DirectAccessUrl requestContentDirectUrl(String nodeId, DirectAccessUrlReq
+    @WebApiParameters({@WebApiParam(name = "nodeId", title = "The unique id of Execution Job", description = "A single nodeId")})
+    public Map<String, Object> calculateFolderSize(String nodeId, Void ignore, Parameters parameters, WithResponse withResponse)
+    {
+        NodeRef nodeRef = nodes.validateNode(nodeId);
+        QName qName = nodeService.getType(nodeRef);
+        Map<String, Object> result = new HashMap<>();
+        validatePermissions(nodeRef, nodeId);
+
+        if(!FOLDER.equalsIgnoreCase(qName.getLocalName()))
+        {
+            throw new InvalidNodeTypeException(INVALID_NODEID);
+        }
+
+        try
+        {
+            FolderSizeImpl folderSizeImpl = new FolderSizeImpl(actionService);
+            return folderSizeImpl.executingAsynchronousFolderAction(nodeRef, defaultItems, result, simpleCache);
+        }
+        catch (Exception alfrescoRuntimeError)
+        {
+            LOG.error("Exception occurred in NodesEntityResource:calculateFolderSize {}", alfrescoRuntimeError.getMessage());
+            throw new AlfrescoRuntimeException("Exception occurred in NodesEntityResource:calculateFolderSize",alfrescoRuntimeError);
+        }
+    }
+
+    @Override
+    @WebApiDescription(title = "Returns Folder Node Size", description = "Returning a Folder Node Size")
+    @WebApiParameters({@WebApiParam(name = "nodeId", title = "The unique id of Execution Job", description = "A single nodeId")})
+    @BinaryProperties({"size"})
+    public Map<String, Object> getFolderSize(String nodeId) throws EntityNotFoundException
+    {
+        Map<String, Object> result = new HashMap<>();
+        try
+        {
+            LOG.debug("Retrieving OUTPUT from NodeSizeActionExecutor - NodesEntityResource:getFolderSize");
+            NodeRef nodeRef = nodes.validateNode(nodeId);
+            validatePermissions(nodeRef, nodeId);
+            QName qName = nodeService.getType(nodeRef);
+
+            if(!FOLDER.equalsIgnoreCase(qName.getLocalName()))
+            {
+                throw new InvalidNodeTypeException(INVALID_NODEID);
+            }
+
+            Object cachedResult = simpleCache.get(nodeId);
+            if(cachedResult != null)
+            {
+                return getResult(cachedResult, result);
+            }
+            else
+            {
+                result.put(STATUS,NOT_INITIATED);
+            }
+            return result;
+        }
+        catch (Exception ex)
+        {
+            LOG.error("Exception occurred in NodesEntityResource:getFolderSize {}", ex.getMessage());
+            throw ex; // Rethrow with original stack trace
+        }
+    }
+
+    /**
+     * Providing the response from SimpleCache.
+     */
+    private Map<String, Object> getResult(Object outputResult, Map<String, Object> result)
+    {
+        if (outputResult instanceof Map)
```

Comment:
If we just keep the value as Map then we can avoid the checks for the data type.

Reply:
Done